### PR TITLE
Update nvcomp to 3.0.4 (includes API changes)

### PR DESCRIPTION
**Pull request by vuule**
Update the nvCOMP version used for compression/decompression to 3.0.4.

See also:

https://github.com/rapidsai/cudf/pull/13815
https://github.com/rapidsai/rapids-cmake/pull/451

### CODE REVIEW

**Comment by vuule** [comment]
reverted set_scratch_allocators to at least see CI passing with nvcomp 3.0.4, even with missing API

**Comment by vuule** [comment]
omg, there's more
all managers have a new parameter in the constuctors

File: cpp/cmake/fetch_rapids.cmake
Comment (bdice) [inline cpp/cmake/fetch_rapids.cmake:15]: After https://github.com/rapidsai/rapids-cmake/pull/451 is merged, revert these lines (before merging this PR).
```suggestion
```

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:21]: Devcontainer CI will have to remain disabled until cudf is updated with https://github.com/rapidsai/cudf/pull/13815, due to the conflicting `nvcomp` dependency which creates circular conflicts between cudf and kvikio.

**Review by bdice — APPROVED** [review]
Approving, with two notes about future changes and how we will sort them out.

**Review by madsbk — APPROVED** [review]
Looks good, thanks @vuule

**Review by bdice — APPROVED** [review]
Re-approving since this was in draft before.

**Comment by bdice** [comment]
@madsbk Can you re-approve on behalf of Python codeowners? This was in draft so your prior approval won't count, I think.

**Review by raydouglass — APPROVED** [review]

File: cpp/cmake/fetch_rapids.cmake
Comment (bdice) [reply cpp/cmake/fetch_rapids.cmake:15]: I forgot to commit this before merging. It's in https://github.com/rapidsai/kvikio/pull/316.

**Pull request merged**
